src/pages/guides/data-fetching.md

```diff
@@ -3,27 +3,53 @@ layout: ~/layouts/Main.astro
 title: Data Fetching
 ---
 
-Astro components and pages can fetch remote data to help generate your pages. Astro provides two different tools to pages to help you do this: **fetch()** and **top-level await.** 
+Astro components and pages can fetch remote data to help generate your pages. Astro provides two different tools to pages to help you do this: **fetch()** and **top-level await.**
 
 ### `fetch()`
 
-Astro pages have access to the global `fetch()` function in their setup script. `fetch()` is a native JavaScript API ([MDN](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch)) that lets you make HTTP requests for things like APIs and resources. 
+Astro pages have access to the global `fetch()` function in their setup script. `fetch()` is a native JavaScript API ([MDN](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch)) that lets you make HTTP requests for things like APIs and resources.
 
 Even though Astro component scripts run inside of Node.js (and not in the browser) Astro provides this native API so that you can fetch data at page build time.
 
-
 ```astro
+// Movies.astro
 ---
-const response = await fetch('http://example.com/movies.json');
+const response = await fetch('https://example.com/movies.json');
 const data = await response.json();
 // Remember: Astro component scripts log to the CLI
 console.log(data);
 ---
 <!-- Output the result to the page -->
 <div>{JSON.stringify(data)}</div>
 ```
+
 ### Top-level await
 
 `await` is another native JavaScript feature that lets you await the response of some asynchronous promise ([MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/await)). Astro supports `await` in the top-level of your component script.
 
 **Important:** These are not yet available inside of non-page Astro components. Instead, do all of your data loading inside of your pages, and then pass them to your components as props.
+
+### `node-fetch`
+
+If you want to use `fetch()` in a non-astro component, use the [`node-fetch`](https://github.com/node-fetch/node-fetch) library:
```

Comment:
there should be a note here somewhere about what happens if you move this component to the client:
- will node-fetch keep working? I don't think it will. That's fine until we can fix this issue for real and make `fetch` available to all.
- that console log will run anywhere, not just in the CLI

Reply:
It looks like we might want to recommend a different library 🤔 Or mention this blurb from the `node-fetch` README at least:

> Instead of implementing XMLHttpRequest in Node.js to run browser-specific [Fetch polyfill](https://github.com/github/fetch), why not go from native http to fetch API directly? Hence, node-fetch, minimal code for a window.fetch compatible API on Node.js runtime.
>
> See Jason Miller's [isomorphic-unfetch](https://www.npmjs.com/package/isomorphic-unfetch) or Leonardo Quixada's [cross-fetch](https://github.com/lquixada/cross-fetch) for isomorphic usage (exports node-fetch for server-side, whatwg-fetch for client-side).
from https://github.com/node-fetch/node-fetch#motivation

So I'll just add a qualifying comment that you'll need to pay attention to whether the tool is in the buildtime or runtime for now, but maybe this should be different 🤔

Reply:
<img width="963" alt="image" src="https://user-images.githubusercontent.com/10626596/125552389-5be434e6-d60c-412c-849a-aa89ae867a5e.png">

Yup just confirmed that `node-fetch` is not isomorphic -- I'll just quote the README as I did above then..